### PR TITLE
REGRESSION(268320@main): WebDriver fullscreen/input type=file tests failing

### PR DESCRIPTION
**Pull request by aprotyas**
#### 0ff99ca2803f94bb9bf6c231975fcd972cde2620
<pre>
REGRESSION(268320@main): WebDriver fullscreen/input type=file tests failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=263887">https://bugs.webkit.org/show_bug.cgi?id=263887</a>
<a href="https://rdar.apple.com/117683701">rdar://117683701</a>

Reviewed by NOBODY (OOPS!).

This commit introduces the ability to force user gesture on script
execution in WebAutomationSession, so we can be deliberate about which
Automation Protocol script requires user gesture, rather than the older
behavior of backing every script execution with a user gesture.

* Source/WebKit/UIProcess/Automation/Automation.json:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::evaluateJavaScriptFunction):
* Source/WebKit/UIProcess/Automation/WebAutomationSession.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::WebAutomationSessionProxy::evaluateJavaScriptFunction):
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.js:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.messages.in:
</pre>

### CODE REVIEW

File: Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp [hunk 1]
```diff
@@ -414,14 +414,19 @@ void WebAutomationSessionProxy::evaluateJavaScriptFunction(WebCore::PageIdentifi
...
         JSValueMakeNumber(context, frameID.processIdentifier().toUInt64()),
         JSValueMakeNumber(context, callbackID),
         JSObjectMakeFunctionWithCallback(context, nullptr, evaluateJavaScriptCallback),
         JSValueMakeNumber(context, callbackTimeout.value_or(-1))
     };
 
-    callPropertyFunction(context, scriptObject, "evaluateJavaScriptFunction"_s, std::size(functionArguments), functionArguments, &exception);
+    if (forceUserGesture) {
+        WebCore::UserGestureIndicator gestureIndicator { WebCore::ProcessingUserGesture, frame->coreLocalFrame()->document() };
+        callPropertyFunction(context, scriptObject, "evaluateJavaScriptFunction"_s, std::size(functionArguments), functionArguments, &exception);
```
Comment (nt1m) [inline Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp]: You can use `emplace()` to avoid duping `callPropertyFunction`

Comment (aprotyas) [reply Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp]: Hmm... sorry I don't follow. Use `emplace()` how?

Comment (nt1m) [reply Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp]:
```
std::optional<WebCore::UserGestureIndicator> gestureIndicator;
if (forceUserGesture)
    gestureIndicator.emplace({ WebCore::ProcessingUserGesture, frame->coreLocalFrame()->document() });
callPropertyFunction(context, scriptObject, "evaluateJavaScriptFunction"_s, std::size(functionArguments), functionArguments, &exception);

Comment (aprotyas) [reply Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp]: Makes sense, I'll change to this. Thanks!

File: Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp [hunk 2]
```diff
@@ -414,13 +414,16 @@ void WebAutomationSessionProxy::evaluateJavaScriptFunction(WebCore::PageIdentifi
...
+        JSValueMakeBoolean(context, forceUserGesture),
         JSValueMakeNumber(context, frameID.object().toUInt64()),
         JSValueMakeNumber(context, frameID.processIdentifier().toUInt64()),
         JSValueMakeNumber(context, callbackID),
         JSObjectMakeFunctionWithCallback(context, nullptr, evaluateJavaScriptCallback),
         JSValueMakeNumber(context, callbackTimeout.value_or(-1))
     };
 
+    auto isProcessingUserGesture = forceUserGesture ? std::optional { WebCore::IsProcessingUserGesture::Yes } : std::nullopt;
+    WebCore::UserGestureIndicator gestureIndicator { isProcessingUserGesture, frame->coreLocalFrame()->document() };
```
Comment (nt1m) [inline Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:426]: std::optional + emplace() seems like a better pattern here

Comment (aprotyas) [reply Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:426]: Just a personal preference of mine; I don't like uninitialized variables. It also seems like there's bigger precedence for `IsProcessingGesture::Yes : std::nullopt` than the other pattern!

**Review by nt1m — APPROVED** [review]
r=me

**Review by patrickangle — APPROVED** [review]

**Comment by aprotyas** [comment]
Closing this for now with the hope that we revisit this in the future. Taking https://github.com/WebKit/WebKit/pull/19716 instead.

**Pull request closed without merging**